### ISSUE TITLE
Sway going idle fails to re-start eDP-1

### ISSUE THREAD

**Issue by rtyler** [swaywm/sway#3531]
After rebuilding sway and wlroots this morning, it seems that my laptop display no longer wishes to reactivate after idling.

Configuration below, will try to nail down some reproduction steps.

* Sway Version: sway version 1.0-beta.2-283-g7c27d73b (Jan 28 2019, branch 'master')

* Debug Log: https://gist.github.com/rtyler/a3a4abf2ad71a2e4335f5a3b1c57e6a4

* Configuration File:

The relevant bits which are resulting in the monitors switching off.
```
### Idle configuration
exec swayidle \
    timeout 300 'swaylock -c 000000' \
    timeout 600 'swaymsg "output * dpms off"' \
       resume 'swaymsg "output * dpms on"' \
    before-sleep 'swaylock -c 000000'
```

```
Output eDP-1 'Sharp Corporation 0x144A 0x00000000'
  Current mode: 3200x1800 @ 59.981998 Hz
  Position: 1920,0
  Scale factor: 1.700000
  Transform: normal
  Workspace: 1
  Available modes:
    3200x1800 @ 47.985001 Hz
    3200x1800 @ 59.981998 Hz

Output DP-1 'Hewlett Packard HP ENVY 32 CNC4450CF3' (focused)
  Current mode: 1920x1200 @ 59.950001 Hz
  Position: 0,0
  Scale factor: 1.000000
  Transform: normal
  Workspace: 3
  Available modes:
    720x400 @ 70.082001 Hz
    640x480 @ 59.939999 Hz
    640x480 @ 59.939999 Hz
    640x480 @ 60.000000 Hz
    720x480 @ 59.939999 Hz
    720x480 @ 59.939999 Hz
    720x480 @ 60.000000 Hz
    720x480 @ 60.000000 Hz
    720x576 @ 50.000000 Hz
    720x576 @ 50.000000 Hz
    800x600 @ 60.317001 Hz
    1024x768 @ 60.004002 Hz
    1280x720 @ 50.000000 Hz
    1280x720 @ 59.939999 Hz
    1280x720 @ 60.000000 Hz
    1280x720 @ 60.000000 Hz
    1440x900 @ 59.901001 Hz
    1280x1024 @ 60.020000 Hz
    1600x900 @ 60.000000 Hz
    1680x1050 @ 59.882999 Hz
    1600x1200 @ 60.000000 Hz
    1920x1080 @ 29.969999 Hz
    1920x1080 @ 30.000000 Hz
    1920x1080 @ 50.000000 Hz
    1920x1080 @ 50.000000 Hz
    1920x1080 @ 59.939999 Hz
    1920x1080 @ 60.000000 Hz
    1920x1080 @ 60.000000 Hz
    1920x1200 @ 59.950001 Hz
````

```
2019-01-28 14:04:36 - [backend/drm/drm.c:521] Cannot modeset 'eDP-1': no CRTC for this connector
2019-01-28 14:04:36 - [sway/config/output.c:220] Failed to modeset output eDP-1
```

**Comment by emersion** [comment]
Can you run `echo 0xFF | sudo tee /sys/module/drm/parameters/debug`, then trigger the bug, then `echo 0x00 | sudo tee /sys/module/drm/parameters/debug`, and finally upload your `dmesg` log? 

Also, can you post a full debug log, not a truncated one? (The one you link is missing the startup part)

```
2019-01-28 13:57:23 - [backend/drm/atomic.c:55] DP-1: Atomic commit failed (pageflip): Invalid argument
2019-01-28 13:57:23 - [backend/drm/atomic.c:62] DP-1: Atomic commit without new changes failed (pageflip): Invalid argument
2019-01-28 13:57:23 - [backend/drm/atomic.c:55] eDP-1: Atomic commit failed (pageflip): Invalid argument
2019-01-28 13:57:23 - [backend/drm/atomic.c:62] eDP-1: Atomic commit without new changes failed (pageflip): Invalid argument
```

**Comment by rtyler** [comment]
@emersion I can do, but will need to be later today once I'm done working :smile_cat:

**Comment by JonnyMako** [comment]
@rtyler does reloading sway bring the display back? I've seen something similar with external displays (sometimes when hotpulgging). In my case reloading activates all displays.

**Comment by rtyler** [comment]
@JonnyMako For what it is worth, no reloading sway does not change anything.

**Comment by RedSoxFan** [comment]
Can you please test https://github.com/swaywm/sway/pull/3689? I think it should fix the issue since #3688 appears to be a duplicate of this one.

**Comment by rtyler** [comment]
Yeah, I was going to come in this weekend and suggest this should be closed. Since jumping to wlroots 0.3, and sway 1.0rc1, I have not seen this.
